sdk/client/client.go

```diff
@@ -513,12 +513,17 @@ func (c *Client) Execute(ctx context.Context, req *Request) (*Response, error) {
 
 	// Determine whether response status is valid
 	if !containsStatusCode(req.ValidStatusCodes, resp.StatusCode) {
-		// The status code didn't match, but we also need to check the ValidStatusFUnc, if provided
+		// The status code didn't match, but we also need to check the ValidStatusFunc, if provided
 		// Note that the odata argument here is a best-effort and may be nil
 		if f := req.ValidStatusFunc; f != nil && f(resp.Response, resp.OData) {
 			return resp, nil
 		}
 
+		status := fmt.Sprintf("%d", resp.StatusCode)
+		if statusText := http.StatusText(resp.StatusCode); statusText != "" {
```

Comment:
Azure doesn't always use the Status Code to mean what the HTTP status means, but the HTTP Status Text is returned along side this, so we should be able to make this:

```suggestion
		if statusText := resp.Status; statusText != nil {
```